src/Fixture.hpp

```diff
@@ -39,13 +53,18 @@ namespace iodrivers_base
      *       // Optional: open here
      *       // driver.openURI("test://")
      *    }
+     *    
+     *    virtual void TearDown()
+     *    {
+     *      tearDownMock();
```

Comment:
I'm really not very fond of requiring the user to do this himself.

I've had a look at the Google Mock code. The [reporting in post-test expectation verification](https://github.com/google/googletest/blob/master/googlemock/src/gmock-spec-builders.cc#L466) is done by accessing an `Expect` function that GMock itself defines. This function is considered internal though, so it might be a problem using it down the line.

The "proper" way I guess would be to build and register expectation objects in the GMock case but that means having a widely different codepath on GTest and boost test ... I think that using Expect in the destructor to report errors would be enough for now.